packages/core/src/utils/extractPrebindingData.ts

```diff
@@ -0,0 +1,179 @@
+    };
+
+    prebindingDataByPatternId[patternId] = prebindingData;
+  }
+
+  return prebindingDataByPatternId;
+};
+
+export const generateDefaultDataSourceForPrebindingDefinition = (
+  prebindingDefinitions: ExperienceComponentSettings['prebindingDefinitions'] = [],
+) => {
+  if (
+    !prebindingDefinitions ||
+    !Array.isArray(prebindingDefinitions) ||
+    !prebindingDefinitions.length
+  ) {
+    return { dataSource: {}, parameters: {} };
+  }
+
+  const prebindingDefinition = prebindingDefinitions[0];
+
+  const dataSource: ExperienceDataSource = {};
+  const parameters: Record<string, Parameter> = {};
+
+  for (const [parameterId, parameterDefinition] of Object.entries(
+    prebindingDefinition.parameterDefinitions ?? {},
+  )) {
+    if (parameterDefinition.defaultSource && isLink(parameterDefinition.defaultSource.link)) {
+      const dataSourceKey = generateRandomId(7);
+      dataSource[dataSourceKey] = parameterDefinition.defaultSource.link;
+      parameters[parameterId] = {
+        type: 'BoundValue',
+        path: `/${dataSourceKey}`,
+      };
+    }
+  }
+  return {
+    dataSource,
+    parameters,
+  };
+};
+
+export function getTargetPatternMappingForParameter({
+  fetchedPatterns,
+  prebindingDataByPatternId,
+  patternNodeDefinitionId,
+  parameterId,
+}: {
+  fetchedPatterns: Array<ExperienceEntry>;
+  prebindingDataByPatternId: Record<string, PrebindingData>;
+  patternNodeDefinitionId: string;
+  parameterId: string;
+}) {
+  const patternPrebindingData = prebindingDataByPatternId[patternNodeDefinitionId];
+
+  if (!patternPrebindingData) return undefined;
+  if (patternPrebindingData.parameterIds.includes(parameterId)) {
+    if (patternPrebindingData.nativeParameterId === parameterId) {
+      if (!patternPrebindingData.variableMappings) return undefined;
+      return Object.fromEntries(
+        Object.entries(patternPrebindingData.variableMappings).filter(
+          ([, mapping]) => mapping.parameterId === parameterId,
+        ),
+      );
+    } else {
+      const parameterDefinition = patternPrebindingData.parameterDefinitions[parameterId];
+      if (!parameterDefinition || !parameterDefinition.passToNodes) return undefined;
+
+      const patternEntry = fetchedPatterns.find(
+        (entry) => entry.sys.id === patternNodeDefinitionId,
+      );
+      if (!patternEntry) return undefined;
+
+      let nestedPatternNode: ComponentTreeNode | undefined;
+
+      treeVisit(
+        {
+          definitionId: 'root',
+          parameters: {},
+          children: patternEntry.fields.componentTree.children,
+        } as ComponentTreeNode,
+        (node) => {
+          if (node.id === parameterDefinition.passToNodes?.[0].nodeId) {
+            nestedPatternNode = node;
+          }
+
+          return undefined;
+        },
+      );
+
+      if (!nestedPatternNode) {
+        return undefined;
+      }
+
+      return getTargetPatternMappingForParameter({
```

Comment:
Technically, if parent pattern limits the allowed content types then we should remove that content type from the returned mappings' `pathsByContentType` field.
Although I am not sure if at this point (preview/delivery) we should be adding this extra complexity. If the user somehow managed to add the wrong entry as the parameter source, we could still render it. Maybe we should be guarding against this case only in editor mode? 🤔 
Curious what you think?